### ISSUE TITLE
param:Optional[x] = None type hint behaves differently than Robot Framework

### ISSUE THREAD

**Issue by mkorpela** [robotframework/PythonLibCore#64]
Please add the following "hacky" Optional removal to https://github.com/robotframework/PythonLibCore/blob/9fcb620841b1b411d43fd710bade64f8d0a9020b/src/robotlibcore.py#L232

     def _remove_optional_none_type_hints(self, type_hints, defaults):
        # If argument has None as a default, typing.get_type_hints adds
        # optional None to the information it returns. We don't want that.
        for arg in defaults:
            if defaults[arg] is None and arg in type_hints:
                type_ = type_hints[arg]
                if self._is_union(type_):
                    try:
                        types = type_.__args__
                    except AttributeError:
                        # Python 3.5.2's typing uses __union_params__ instead
                        # of __args__. This block can likely be safely removed
                        # when Python 3.5 support is dropped
                        types = type_.__union_params__
                    if len(types) == 2 and types[1] is type(None):
                        type_hints[arg] = types[0]

https://github.com/robotframework/robotframework/blob/5dc9affd4aee31703a58ad105ed49579b1ca7067/src/robot/running/arguments/argumentparser.py#L101

**Comment by aaltat** [comment]
So if my keyword looks like this:
```
    def optional_none(self, arg: Optional[str] = None):
        pass
```
Then the types returned by PLC should look like this: `{'arg': Union[str, None]}`, or did I misunderstood something? Like in #66 ?

**Comment by mkorpela** [comment]
And defaults should return None

**Comment by aaltat** [comment]
Errr, need more work to understand what you mean?

**Comment by mkorpela** [comment]
So this is wanted behaviour:

suite.robot:
```
*** Settings ***
Library  lib.py

*** Test Cases ***
Testing
    kw  ok

Testing 2
    kw  fail
```

and lib.py:
```
from enum import Enum
from typing import Optional

penum = Enum("penum", "ok")

def kw(param: Optional[penum] = None):
    print(f"Ok? {param}")
```

Here robot will say:
"Testing 2                                                             | FAIL |
ValueError: Argument 'param' got value 'fail' that cannot be converted to penum: penum does not have member 'fail'. Available: 'ok'"

But with PythonLibCore.. continuing in a bit.

**Comment by mkorpela** [comment]
```
from enum import Enum
from typing import Optional
from robotlibcore import DynamicCore, keyword

penum = Enum("penum", "ok")


class KwLib(object):

  @keyword
  def kw(self, param: Optional[penum] = None):
    print(f"Ok? {param}")

class lib(DynamicCore):

    def __init__(self):
       DynamicCore.__init__(self, [KwLib()])
```

This just passes -> that is wrong

**Comment by aaltat** [comment]
Good example, have to write test for it.

**Comment by aaltat** [comment]
[Tested](https://github.com/robotframework/PythonLibCore/pull/66/files#diff-9f889ccd6df6cec4d69d01c9226bc0a9R66) and this is fixed by #66
